source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.cpp

```diff
@@ -0,0 +1,638 @@
+	allControls_->addControl(slowThresholdControl_);
+	allControls_->addControl(peakingTimeControl_);
+	allControls_->addControl(fastChannelPeakingTimeControl_);
+	allControls_->addControl(roi1LowIndexControl_);
+	allControls_->addControl(roi1HighIndexControl_);
+	allControls_->addControl(roi2LowIndexControl_);
+	allControls_->addControl(roi2HighIndexControl_);
+	allControls_->addControl(roi3LowIndexControl_);
+	allControls_->addControl(roi3HighIndexControl_);
+	allControls_->addControl(roi4LowIndexControl_);
+	allControls_->addControl(roi4HighIndexControl_);
+	allControls_->addControl(roi5LowIndexControl_);
+	allControls_->addControl(roi5HighIndexControl_);
+	allControls_->addControl(roi6LowIndexControl_);
+	allControls_->addControl(roi6HighIndexControl_);
+	allControls_->addControl(roi7LowIndexControl_);
+	allControls_->addControl(roi7HighIndexControl_);
+	allControls_->addControl(roi8LowIndexControl_);
+	allControls_->addControl(roi8HighIndexControl_);
+	/**/
+
+	connect(this, SIGNAL(configurationValuesUpdate(AmptekConfigurationData)), this, SLOT(onConfigurationValuesUpdate(AmptekConfigurationData)));
+
+	connect(allControls_, SIGNAL(connected(bool)), this, SLOT(onAllControlsConnected(bool)));
+	connect(allControls_, SIGNAL(controlSetTimedOut()), this, SLOT(onAllControlsTimedOut()));
+
+	connect(startDwellControl_, SIGNAL(valueChanged(double)), this, SLOT(onStartDwellControlValueChange(double)));
+	connect(stopDwellControl_, SIGNAL(valueChanged(double)), this, SLOT(onStopDwellControlValueChange(double)));
+	connect(clearSpectrumControl_, SIGNAL(valueChanged(double)), this, SLOT(onClearSpectrumControlValueChange(double)));
+	connect(dwellTimeControl_, SIGNAL(valueChanged(double)), this, SLOT(onDwellTimeControlValueChange(double)));
+	connect(dwellModeControl_, SIGNAL(valueChanged(double)), this, SLOT(onDwellModeControlValueChange(double)));
+
+	/**/
+	connect(highVoltageControl_, SIGNAL(valueChanged(double)), this, SLOT(onHighVoltageControlValueChanged(double)));
+	connect(totalGainControl_, SIGNAL(valueChanged(double)), this, SLOT(onTotalGainControlValueChanged(double)));
+	connect(pileUpRejectionControl_, SIGNAL(valueChanged(double)), this, SLOT(onPileUpRejectionControlValueChanged(double)));
+	connect(thermoelectricCoolerControl_, SIGNAL(valueChanged(double)), this, SLOT(onThermoelectricCoolerControlValueChanged(double)));
+	connect(fastThresholdControl_, SIGNAL(valueChanged(double)), this, SLOT(onFastThresholdControlValueChanged(double)));
+	connect(slowThresholdControl_, SIGNAL(valueChanged(double)), this, SLOT(onSlowThresholdControlValueChanged(double)));
+	connect(peakingTimeControl_, SIGNAL(valueChanged(double)), this, SLOT(onPeakingTimeControlValueChanged(double)));
+	connect(fastChannelPeakingTimeControl_, SIGNAL(valueChanged(double)), this, SLOT(onFastChannelPeakingTimeControlValueChanged(double)));
+	connect(roi1LowIndexControl_, SIGNAL(valueChanged(double)), this, SLOT(onROI1LowIndexControlValueChanged(double)));
```

Comment:
This might be a better question for @davidChevrier, but do we care about these ROIs anymore?